app/src/main/java/com/uwcs446/socialsports/domain/match/MatchRepository.kt

```
@@ -8,13 +8,13 @@ interface MatchRepository {
 
     val matchesByHost: LiveData<Pair<String, List<Match>>>
 
-    suspend fun fetchExploreMatches(sport: Sport): List<Match>?
+    suspend fun fetchExploreMatches(sport: Sport): List<Match>
 
-    suspend fun findAllByHost(hostId: String): List<Match>?
+    suspend fun findAllByHost(hostId: String): List<Match>
 
-    suspend fun findJoinedByUser(userId: String): List<Match>?
+    suspend fun findJoinedByUser(userId: String): List<Match>
 
-    suspend fun findPastWithUser(userId: String): List<Match>?
+    suspend fun findPastWithUser(userId: String): List<Match>
 
     suspend fun fetchMatchById(matchId: String): Match?
```

Should we remove nullness here too?

---

Good question - for the other functions, I could remove nullness since we can always return a list of matches (it could be an empty list).

For the individual `byId` functions (this also addresses your first comment), it's possible that a user or match doesn't exist, so it'll return null.

To return non-nullable `User` and `Match`,  I guess we could also throw an exception if a user or match isn't found, but I'll leave that consideration for a different PR